### PR TITLE
feat(solid-query): add plugin

### PR DESCRIPTION
**Pull request by qlaffont**
🚨 **IMPORTANT: Please do not create a Pull Request without creating an issue first.**

https://github.com/dotansimha/graphql-code-generator-community/issues/469

## Description

Implement Solid-query generator

Related #469

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Use existing tests for react-query and change it accordingly to solid query documentation https://tanstack.com/query/latest/docs/solid/overview

## Checklist:

- [X] I have followed the
      [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the
      style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by qlaffont** [comment]
@saihaj  / @TkDodo can you review my PR please ? :)

**Comment by qlaffont** [comment]
@dotansimha :wave:

**Comment by saihaj** [comment]
Hey! Sorry for the delay. Assigning myself for review so I can get it done this week

**Comment by qlaffont** [comment]
@saihaj any update ?

**Comment by qlaffont** [comment]
@dotansimha  ?

**Review by saihaj — APPROVED** [review]
we would also need to update codegen website in the codegen repo to this package

File: packages/plugins/typescript/solid-query/package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "name": "@graphql-codegen/typescript-solid-query",
+  "version": "1.0.0",
```
Comment (saihaj) [inline packages/plugins/typescript/solid-query/package.json:3]:
```suggestion
  "version": "0.0.1",
```

File: packages/plugins/typescript/solid-query/package.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  "type": "module",
+  "description": "GraphQL Code Generator plugin for generating a ready-to-use Solid-Query Hooks based on GraphQL operations",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/dotansimha/graphql-code-generator-community.git",
+    "directory": "packages/plugins/typescript/solid-query"
+  },
+  "license": "MIT",
+  "engines": {
+    "node": ">= 16.0.0"
```
Comment (saihaj) [inline packages/plugins/typescript/solid-query/package.json:13]: let's do `18`
```suggestion
    "node": ">= 18.0.0"
```

**Review by saihaj — COMMENTED** [review]
actually before we merge. make that version change in package json I proposed and then run `yarn changeset` and create a `major` then it will automatically work with your deploy setup

**Comment by qlaffont** [comment]
@saihaj 
```
@qlaffont ➜ /workspaces/graphql-code-generator-community (main) $ yarn changeset major
yarn run v1.22.19
error Command "changeset" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@qlaffont ➜ /workspaces/graphql-code-generator-community (main) $ yarn
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
error https://registry.yarnpkg.com/typescript/-/typescript-4.9.5.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "ENOENT: no such file or directory, stat '/home/codespace/.cache/yarn/v6/npm-typescript-4.9.5-095979f9bcc0d09da324d58d03ce8f8374cbe65a-integrity/node_modules/typescript/lib/typescript.js'"
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by qlaffont** [comment]
@saihaj ready to merge !

**Comment by qlaffont** [comment]
@saihaj any update ?

File: packages/plugins/typescript/solid-query/.changeset/README.md
Comment (saihaj) [inline packages/plugins/typescript/solid-query/.changeset/README.md:1]: it shouldn't create this new directory

Comment (qlaffont) [reply packages/plugins/typescript/solid-query/.changeset/README.md:1]: @saihaj understand but what can I do ? I delete the folder ?

Comment (saihaj) [reply packages/plugins/typescript/solid-query/.changeset/README.md:1]: yeah delete. In the root of this when you do `yarn changeset` it should work give this is part of the `yarn workspace `

Comment (qlaffont) [reply packages/plugins/typescript/solid-query/.changeset/README.md:1]: @saihaj changes done

**Comment by qlaffont** [comment]
@saihaj changes done

**Comment by qlaffont** [comment]
@saihaj any update ?

**Comment by qlaffont** [comment]
@saihaj any update ?

**Comment by qlaffont** [comment]
@saihaj any update ?

**Comment by qlaffont** [comment]
@dotansimha @saihaj 👋

**Comment by qlaffont** [comment]
@dotansimha @saihaj 👋

**Comment by qlaffont** [comment]
@dotansimha @saihaj 👋

**Review by saihaj — APPROVED** [review]
sorry for the delay

**Pull request merged**